server/src/main/java/io/spine/server/BoundedContext.java

```diff
@@ -350,13 +352,42 @@ public CommandErrorHandler createCommandErrorHandler() {
     public Optional<Repository> findRepository(Class<? extends Message> entityStateClass) {
         // See if there is a repository for this state at all.
         if (!guard.hasRepository(entityStateClass)) {
-            throw newIllegalStateException("No repository found for the the entity state class %s",
+            throw newIllegalStateException("No repository found for the entity state class `%s`.",
                                            entityStateClass.getName());
         }
         Optional<Repository> repository = guard.repositoryFor(entityStateClass);
         return repository;
     }
 
+    /**
+     * Verifies if there is a repository managing the passed entity class is registered
+     * with this Bounded Context.
+     *
+     * <p>This method does not take into account visibility of entity states.
+     *
+     * @see #findRepository(Class)
+     */
+    @VisibleForTesting
+    public boolean hasRepositoryOf(Class<? extends Entity<?, ?>> entityClass) {
+        EntityClass<? extends Entity<?, ?>> cls = EntityClass.asEntityClass(entityClass);
+        boolean result = guard.hasRepository(cls.stateClass());
+        return result;
+    }
+
+    /**
+     * Verifies if there a repository managing entities with the state of the passed class
+     * registered with this Bounded Context.
+     *
+     * <p>This method does not take into account visibility of entity states.
+     *
+     * @see #findRepository(Class)
+     */
+    @VisibleForTesting
+    public boolean hasRepositoryOfEntityWithState(Class<? extends Message> stateClass) {
```

Comment:
This is too long. Also, the default behaviour for the `findRepository` method is to look up by the entity state type, thus the `has...` method looks up by the entity state type should have the shorter name.